src/main/java/com/manywho/services/saml/services/AuthenticationService.java

```diff
@@ -25,20 +28,30 @@ public AuthenticatedWhoResult createAuthenticatedWhoResultWithError(String error
     }
 
     public AuthenticatedWhoResult createAuthenticatedWhoResult(ApplicationConfiguration configuration, SamlResponseHandler response) throws Exception {
+        return createAuthenticatedWhoResult(configuration, response, true);
+    }
+
+    public AuthenticatedWhoResult createAuthenticatedWhoResult(ApplicationConfiguration configuration, SamlResponseHandler response, boolean validate) throws Exception {
 
-        if (response.isValid() == false) {
+        if (validate == true && response.isValid() == false) {
             return createAuthenticatedWhoResultWithError(response.getError());
         }
 
         String jwtToken;
 
         try {
+            LocalDateTime notAfter = response.getResponse().getSessionNotAfter();
```

Comment:
this logic could be added inside the method jwtService.sign to keep it consistent to how notAfter, and notBefore have been implemented